tools/extra/ras/main.c

```diff
@@ -834,6 +835,7 @@ fpga_result clear_port_errors(fpga_handle afu_handle)
 	result = fpgaObjectRead64(port_error_object, &value, 0);
 	if (result != FPGA_OK) {
 		OPAE_ERR("Failed to Read Object ");
+		resval = (result != FPGA_OK) ? result : resval;
```

Comment:
If we're falling into this block because of the condition above `result != FPGA_OK` then it is not necessary to do the check again in the ?: operator. Because you're not using result in the error message when `fpgaDestroyObject` fails, you can call the destroy function without assigning it to a variable but instead just checking the result directly. This can avoid having two different result variables.